source/plugins/commands.rst

```diff
@@ -22,39 +141,74 @@ The first step is to create a class for the command. The class has to implement
 
     import com.google.common.base.Optional;
 
+    // Simple command that sends a message to all players: /broadcast <message>
     public class MyBroadcastCommand implements CommandCallable {
 
         private final Server server;
-        private final Optional<String> desc = Optional.of("Displays a message to all players");
-        private final Optional<String> help = Optional.of("Displays a message to all players. It has no color support!");
+
+        private final Optional<String> desc = Optional.of(
+            "Displays a message to all players");
+
+        private final Optional<String> help = Optional.of(
+            "Displays a message to all players. It has no color support!");
 
         public MyBroadcastCommand(Server server) {
             this.server = server;
         }
 
-        public boolean call(CommandSource source, String arguments, List<String> parents) throws CommandException {
-            server.broadcastMessage(Messages.of(arguments));
+        // Called when a CommandSource (player, console, ...) executes the command
+        public boolean call(CommandSource source, String arguments,
+            List<String> parents) throws CommandException {
+
+            // Creates a message from the command arguments
+            Message message = Messages.of(arguments);
+
+            // Sends the message to all players
+            this.server.broadcastMessage(message);
+
+            // Indicates that the command was successful
             return true;
         }
 
+        // Tests if the CommandSource is permitted to run this command
         public boolean testPermission(CommandSource source) {
-            return source.hasPermission("example.exampleCommand");
+            return source.hasPermission("example.broadcast");
         }
 
+        // Displayed when a player uses /help, if the help system supports it
         public Optional<String> getShortDescription() {
             return desc;
         }
 
+        // A longer help text
         public Optional<String> getHelp() {
             return help;
         }
 
+        // A string explaining the arguments of the command
         public String getUsage() {
-            return "/<command> <message>";
+            return "<message>";
         }
 
-        public List<String> getSuggestions(CommandSource source, String arguments) throws CommandException {
-            return Collections.emptyList();
+        // A list of suggestions when a player presses TAB to complete the command
+        public List<String> getSuggestions(CommandSource source, String arguments)
```

Comment:
Thats's right. It would also be better to have a static list of predefined suggestions (predefindedSuggestions = {"Hello World","Welcome","Hi!"}), and then do a check like this:
```java
String argumentsLC = arguments.toLowerCase();
for(String s : predefinedSuggestions) {
    if(s.toLowerCase().startsWith(argumentsLC) suggestions.add(s);
}
return suggestions;
```